samples/cpp/intelligent_scissors.cpp

```diff
@@ -0,0 +1,209 @@
+                        continue;
+                    if (expand.at<uchar>(ty, tx) == 0)
+                    {
+                        Point q = Point(tx, ty);
+                        float cost = cost_map.at<float>(p) + local_cost(p, q);
+                        if (processed.at<uchar>(q) == 1 && cost < cost_map.at<float>(q))
+                        {
+                            removed.at<uchar>(q) = 1;
+                        }
+                        if (processed.at<uchar>(q) == 0)
+                        {
+                            cost_map.at<float>(q) = cost;
+                            hit_map_x.at<int>(q)= p.x;
+                            hit_map_y.at<int>(q) = p.y;
+                            processed.at<uchar>(q) = 1;
+                            Pix val;
+                            val.cost = cost_map.at<float>(q);
+                            val.next_point = q;
+                            L.push(val);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+static void onMouse(int event, int x, int y, int , void *)
+{
+    if (event == EVENT_LBUTTONDOWN)
+    {
+        end = Point(x, y);
+        if (!contours.back().empty())
+        {
+            for(int i = tmp_contour.size() - 1; i >= 0; i--)
+            {
+                contours.back().push_back(tmp_contour[i]);
+            }
+            tmp_contour.clear();
+        }
+        else
+        {
+            contours.back().push_back(end);
+        }
+        find_min_path(end);
+
+        img_render.copyTo(img_pre_render);
+        imshow("lasso", img_render);
+    }
+    else if (event == EVENT_RBUTTONDOWN)
+    {
+        img_pre_render.copyTo(img_render);
+        drawContours(img_pre_render, contours, contours.size() - 1, Scalar(0,255,0), FILLED);
+        addWeighted(img_pre_render, 0.3, img_render, 0.7, 0, img_render);
+        contours.resize(contours.size() + 1);
+        imshow("lasso", img_render);
+    }
+    else if (event == EVENT_MOUSEMOVE && !contours.back().empty())
+    {
+        tmp_contour.clear();
+        img_pre_render.copyTo(img_render);
+        Point val_point = Point(x, y);
+        while (val_point != end)
+        {
+            tmp_contour.push_back(val_point);
+            Point cur = Point(hit_map_x.at<int>(val_point), hit_map_y.at<int>(val_point));
+            line(img_render, val_point, cur, Scalar(255, 0, 0), 2);
+            val_point = cur;
+        }
+        imshow("lasso", img_render);
+    }
+}
+
+const char* keys =
+{
+    "{help h | |}{@image|fruits.jpg|}"
+};
+
+int main( int argc, const char** argv )
+{
+    CommandLineParser parser(argc, argv, keys);
+    parser.about("\nThis program demonstrates implementation of 'intelligent scissors' algorithm\n"
+                 "To start the algorithm select a pixel, press lbm and move a mouse to create a path.\n"
+                 "To stop the algorithm click rbm\n");
```

Comment:
Looks like we need to press once and move. Please rephrase so user can understand that to fixate contour point - click left mouse button. To finish contour - click right mouse button

Reply:
like this?
`    parser.about("\nThis program demonstrates implementation of 'intelligent scissors' algorithm\n"
                 "To start the algorithm select a pixel, click lbm.\nTo create a path move a mouse.\n"
                 "To fix a path click ldm again.\n"
                 "To stop the algorithm click rbm.\n");`

Reply:
`To start the algorithm` -> `To start drawing a new contour`?

`select a pixel, click lbm` -> `click LEFT mouse button`

`To create a path move a mouse.` Can be removed. I think it's more of less intuitive to move a mouse

`ldm` -> `LEFT mouse button`

`To stop the algorithm click rbm.` -> `To finish drawing a contour click RIGHT mouse button`

And please remove `\n` - otherwise it's hard to read in the source code